tests-pytest/action.yml

```diff
@@ -100,13 +100,30 @@ runs:
         python-version: ${{ inputs.python-version }}
         use-cache: ${{ inputs.use-python-cache }}
 
-    - name: "Update pip"
+    - name: "Upgrade pip and disable virtual environment (for poetry only)"
       shell: bash
-      run: python -m pip install -U pip
+      run: |
+        python -m pip install --upgrade pip
+        if grep -q 'build-backend = "poetry\.core\.masonry\.api"' "pyproject.toml"; then
+          # For projects using poetry, do not use a virtual environment.
+          # Poetry uses virtual environments to install its dependencies, but this might
+          # lead to problems if it is not activated prior to executing poetry commands.
+          # Store POETRY_VIRTUALENVS_CREATE=false in the GitHub environment to prevent
+          # poetry from creating a virtual environment.
+          #
+          echo "POETRY_VIRTUALENVS_CREATE=false" >> $GITHUB_ENV
+        fi
 
-    - name: "Install Python library"
+    - name: "Install project (if required)"
+      if: inputs.skip-install == 'false'
       shell: bash
-      run: python -m pip install .
+      run: |
+        if grep -q 'build-backend = "poetry\.core\.masonry\.api"' "pyproject.toml"; then
+          python -m pip install poetry
+          python -m poetry install
```

Comment:
not sure you can call poetry from python. I think it installs as command line tool

Reply:
You are completely right - my bad